# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/flutter/system-dart.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- a/bin/internal/shared.sh
+++ b/bin/internal/shared.sh
@@ -122,7 +122,9 @@ function upgrade_flutter () (
   "$FLUTTER_ROOT/bin/internal/update_engine_version.sh"
 
   local revision="$(git -C "$FLUTTER_ROOT" rev-parse HEAD)"
-  local compilekey="$revision:$FLUTTER_TOOL_ARGS"
+  local dart_version
+  dart_version=$("$DART" --version)
+  local compilekey="$revision:$dart_version:$FLUTTER_TOOL_ARGS"
 
   # Invalidate cache if:
   #  * SNAPSHOT_PATH is not a file, or
